internal/resources/cluster/resource_cluster.go

```diff
@@ -213,20 +238,70 @@ func flattenSpec(spec *clustermodel.VmwareTanzuManageV1alpha1ClusterSpec) (data
 	return []interface{}{flattenSpecData}
 }
 
+func ValidateKubeConfig(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
```

Comment:
The kubeconfig can be validated by checking if the k8s client is nil or not

Reply:
A schema validation checks before even the call is made to TMC, so one the first layer we get the response.

Reply:
you can use : ValidateFunc or ExactlyOneOf  []string attributes from the schema. 
We should not use diff customise diff to handle schema validations

Reply:
I believe than we need to change the same for other resources as well, customise diff has been used in other resources as well

Reply:
if it is only limited to schema validation and something which can be handled via validateFunc we should be using it. 
if other resources needs change, create a task and it can be picked in the future

Reply:
`validateFunc` is being used to check non empty Kube_config inputs. Here we are checking some extra corner cases such as not giving more than one Kube_config input type